src/main/java/edu/cornell/kfs/module/purap/document/service/impl/CuPurapServiceImpl.java

```diff
@@ -215,5 +232,122 @@ private void resetAccountAmount(PurApAccountingLine distributedAccount, KualiDec
         distributedAccount.setAmount(new KualiDecimal(amount));
     }
 
-    
+    /**
+     * This implementation is almost identical to the three-arg version of the method,
+     * except that this one will return a different default limit for qualifying
+     * Federal-fund-related documents.
+     * 
+     * @see edu.cornell.kfs.module.purap.document.service.CuPurapService#getApoLimit(org.kuali.kfs.module.purap.document.PurchasingDocumentBase)
+     */
+    @SuppressWarnings({"rawtypes", "unchecked"})
+    @Override
+    public KualiDecimal getApoLimit(PurchasingDocument purchasingDocument) {
+        Integer vendorContractGeneratedIdentifier = purchasingDocument.getVendorContractGeneratedIdentifier();
+        String chart = purchasingDocument.getChartOfAccountsCode();
+        String org = purchasingDocument.getOrganizationCode();
+        KualiDecimal purchaseOrderTotalLimit = null;
+        
+        if (canUseApoLimitFromOrganizationOrContract(purchasingDocument)) {
```

Comment:
This logic is different from the confluence document that was.  Perhaps the technical documentation should be updated to cover that as well.

Reply:
Good point, I will update the Confluence documentation accordingly.